src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -328,12 +328,31 @@ typedef struct server_vnodes_t {
 } server_vnodes_t;
 static	server_vnodes_t server_vnodes;
 
+/**
+ * @brief
+ * 	This is a function that logs the contents of the list
+ *	headed by 'phead'.
+ *
+ * @param[in] head_str - some string that gets printed spearheading the list.
+ * @param[in] phead - header of the list to be printed.
+ *
+ * @return	void
+ *
+ */
 void
 print_svrattrl_list(char *head_str, pbs_list_head *phead)
 {
 	svrattrl *plist = NULL;
 	int i;
 
+	if ((head_str == NULL) || (phead == NULL) || (log_event_mask == NULL)) {
+		return;
+	}
+
+	if ((*log_event_mask & PBSEVENT_DEBUG3) == 0) {
```

Comment:
I understand why you did it.  Your comment doesn't change my point though.  You are breaking encapsulation of the logging library.  We don't do this anywhere else.  The only place log_event_mask is used is within the the logging library.  

When making changes involved with speed, some testing is required.  I can't tell you how many times I thought that a change would greatly speed up the scheduler only to have it turn out to be some minor little thing.  How long does this loop take?  Before the loop was being done N times.  Now you're only doing it once.  Does it take that much time?  It might be fast enough that making this type of change is not even worthwhile.  Another option is to reevaluate how useful this loop is.  It's at DEBUG3, so it's most likely an unstable interface.  We can remove it if it doesn't provide that much use and is unnecessarily slowing things down.  Keep in mind that if the log event is at DEBUG3, we will still go through the loop.